sys/usb/usbus/dfu/dfu.c

```diff
@@ -0,0 +1,292 @@
+
+    usbus_control_slicer_put_bytes(usbus, (uint8_t *)&if_desc, sizeof(if_desc));
+    return sizeof(usb_desc_if_dfu_t);
+}
+
+static const usbus_handler_driver_t dfu_driver = {
+    .init = _init,
+    .event_handler = _event_handler,
+    .transfer_handler = _transfer_handler,
+    .control_handler = _control_handler,
+};
+
+/* Descriptors */
+static const usbus_descr_gen_funcs_t _dfu_descriptor = {
+    .fmt_post_descriptor = _gen_dfu_descriptor,
+    .fmt_pre_descriptor = NULL,
+    .len = {
+        .fixed_len =  sizeof(usb_desc_if_dfu_t),
+    },
+    .len_type = USBUS_DESCR_LEN_FIXED,
+};
+
+#ifdef MODULE_RIOTBOOT_USB_DFU
+static void _reboot(void *arg)
+{
+    (void)arg;
+    pm_reboot();
+}
+#endif
+
+void usbus_dfu_init(usbus_t *usbus, usbus_dfu_device_t *handler, unsigned mode)
+{
+    DEBUG("DFU: initialization\n");
+    assert(usbus);
+    assert(handler);
+    assert((SLOT0_OFFSET % FLASHPAGE_SIZE) == 0);
+    memset(handler, 0, sizeof(usbus_dfu_device_t));
+    handler->usbus = usbus;
+    handler->handler_ctrl.driver = &dfu_driver;
+    handler->mode = mode;
+    handler->selected_slot = UINT32_MAX;
+    handler->dfu_state = (handler->mode == USB_DFU_PROTOCOL_DFU_MODE) ?
+                USBUS_DFU_STATE_DFU_IDLE : USBUS_DFU_STATE_APP_IDLE;
+
+    usbus_register_event_handler(usbus, (usbus_handler_t *)handler);
+}
+
+static void _init(usbus_t *usbus, usbus_handler_t *handler)
+{
+    usbus_dfu_device_t *dfu = (usbus_dfu_device_t*) handler;
+    /* Set up descriptor generators */
+    dfu->dfu_descr.next = NULL;
+    dfu->dfu_descr.funcs = &_dfu_descriptor;
+    dfu->dfu_descr.arg = dfu;
+
+    /* Configure Interface 0 as control interface */
+    dfu->iface.class = USB_DFU_INTERFACE;
+    dfu->iface.subclass = USB_DFU_SUBCLASS_DFU;
+
+    dfu->iface.protocol = dfu->mode;
+
+    dfu->iface.descr_gen = &dfu->dfu_descr;
+    dfu->iface.handler = handler;
+
+    /* Create needed string descriptor for the interface and its alternate settings */
+    usbus_add_string_descriptor(usbus, &dfu->slot0_str, slot0_name);
+
+    /* Add string descriptor to the interface */
+    dfu->iface.descr = &dfu->slot0_str;
+
+#ifdef MODULE_RIOTBOOT_USB_DFU
+    /* Create needed string descriptor for the alternate settings */
+    usbus_add_string_descriptor(usbus, &dfu->slot1_str, slot1_name);
+
+    /* Add string descriptor to the alternate settings */
+    dfu->iface_alt_slot1.descr = &dfu->slot1_str;
+
+    /* attached alternate settings to their interface */
+    usbus_add_interface_alt(&dfu->iface, &dfu->iface_alt_slot1);
+
+    /* Start xtimer for scheduled reboot after firmware upgrade */
+    xtimer_init();
+#endif
+    /* Add interface to the stack */
+    usbus_add_interface(usbus, &dfu->iface);
+    usbus_handler_set_flag(handler, USBUS_HANDLER_FLAG_RESET);
+}
+
+static void _dfu_class_control_req(usbus_t *usbus, usbus_dfu_device_t *dfu, usb_setup_t *pkt)
+{
+    usbopt_enable_t enable = USBOPT_DISABLE;
```

Comment:
```suggestion
    static const usbopt_enable_t enable = USBOPT_DISABLE;
```

I find `enable` also a bit confusing for something with `USBOPT_DISABLE` as value